### DISCUSSION TITLE
Windows build check

### DISCUSSION THREAD

**Discussion by si618** [hiero-ledger/hiero-mirror-node#2072] [Ideas]
G'day folks,

With the mirror node now building on Windows, I'm wondering if an action or workflow could be added to keep an eye on things?

I appreciate Ubuntu is the preferred environment, and would envisage most Windows developers using docker and WSL2 for deployments, however, having the mirror node reliably build on Windows should be useful as Hedera's decentralization continues and interest grows.

By way of example, [here is the result](https://github.com/si618/hedera-mirror-node/runs/2742511408?check_suite_focus=true) of building on Windows Server 2019 [using this workflow](https://github.com/si618/hedera-mirror-node/commit/b1e413b5d8e0d5a45c9d3f8a0bb75e7fefd4a56b).

A build takes ~5-6 minutes once the maven packages are cached, but not sure if it would be needed on each push; running on (pre)release might suffice?

**Comment by steven-sheehy** [comment]
I think it does have value. On every push might be too much as it's a longer running test. Putting it as a pre-release might be a concern as it could fail the release at the last minute. Perhaps we can have it as a nightly/weekly job instead?

**Comment by steven-sheehy** [comment]
I just noticed your workflow doesn't run the tests. We'd probably want to run those as well to catch Windows specific runtime errors and making it run less often would allow us to.

**Reply by si618** [reply]
Thanks for the prompt response Steven, a nightly or weekly build makes sense and just pushed a change to include tests, but more work is needed to get docker happy.

I'll follow up once the tests are working, hopefully won't be too much of a drama as they are running locally for me.

**Reply by si618** [reply]
Windows containers [aren't currently supported](https://github.com/si618/hedera-mirror-node/runs/2742755282?check_suite_focus=true#step:5:91), same issue described in [more detail](https://github.com/testcontainers/testcontainers-java/issues/2960).

Looks like it's going to be hard to decouple docker, and using a linux container on Windows Server 2019 [appears difficult](https://stackoverflow.com/a/66078265/44540) if [not impossible](https://github.com/MicrosoftDocs/WSL/issues/678) at the moment.

The only way I can see around it would be to use a Windows 10 VM with WSL2, but that's not available on Github, nor [Azure DevOps](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#software), so at this stage the best we can probably do is verify mirror node builds on windows.

**Reply by steven-sheehy** [reply]
Ah, that's too bad. In that case, we can probably just add the build windows to run on every PR as it's only a few minutes.

**Reply by si618** [reply]
Done. First time I've created a PR which [tests itself](https://github.com/hashgraph/hedera-mirror-node/runs/2770275035?check_suite_focus=true) 😀
